lib/pub_sub/poller.rb

```diff
@@ -1,31 +1,38 @@
 module PubSub
   class Poller
-    def initialize(queue_url, verbose = false)
-      @queue_url = queue_url
+    def initialize(verbose = false)
       @verbose = verbose
     end
 
     def poll
-      poller.poll(config) do |message|
-        if @verbose
-          PubSub.logger.info(
-            "PubSub received: #{message.message_id} - #{message.body}"
-          )
+      loop do
+        Breaker.current_breaker.run do
+          poller.poll(idle_timeout: 60) do |message|
+            if @verbose
+              PubSub.logger.info(
+                "PubSub received: #{message.message_id} - #{message.body}"
+              )
+            end
+            Message.new(message.body).process
+          end
         end
-        Message.new(message.body).process
+        Breaker.use_next_breaker
```

Comment:
Coming a bit late to this convo. But curious why `Breaker.use_next_breaker` is being called before the `CB2::BreakerOpen` exception? In case an SNS region goes down?

Also can we move the rescue/switching of the breaker region to the Breaker class? Something along the lines of:

```ruby 
#lib/pub_sub/poller.rb
Breaker.run do |current_region|
   # ...
end

#lib/pub_sub/breaker.rb
module PubSub
  class Breaker
    THREAD_LOCAL_IDENTIFIER = :pub_sub_region

    class << self
      def run(&block)
        begin
          current_breaker.run do
            yield(current_region)
          end 
        rescue CB2::BreakerOpen
          use_next_breaker
          sleep 1
          retry
        end
     end
  end

end 



```